packages/@aws-cdk/aws-codebuild/lib/source.ts

```diff
@@ -636,10 +649,50 @@ class S3Source extends Source {
   }
 }
 
+/**
+ * Common properties between {@link GitHubSource} and {@link GitHubEnterpriseSource}.
+ */
+interface CommonGithubSourceProps extends ThirdPartyGitSourceProps {
+  /**
+   * This parameter is used for the `context` parameter in the GitHub commit status.
+   *
+   * @see https://docs.aws.amazon.com/codebuild/latest/userguide/create-project-cli.html#cli.source.buildstatusconfig.context
+   * @see https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-env-vars.html
+   *
+   * @example "My build #$CODEBUILD_BUILD_NUMBER"
+   * @default "AWS CodeBuild $AWS_REGION ($PROJECT_NAME)"
+   */
+  readonly buildStatusContext?: string
+}
+
+abstract class CommonGithubSource extends ThirdPartyGitSource {
+  private readonly buildStatusContext?: string;
+
+  constructor(props: CommonGithubSourceProps) {
+    super(props);
+    this.buildStatusContext = props.buildStatusContext;
+  }
+
+  public bind(_scope: CoreConstruct, project: IProject): SourceConfig {
+    const superConfig = super.bind(_scope, project);
+    return {
+      sourceProperty: {
+        ...superConfig.sourceProperty,
+        buildStatusConfig: {
+          context: this.buildStatusContext,
+          targetUrl: this.buildStatusUrl,
+        },
```

Comment:
We need to render this property only if at least one of `this.buildStatusContext` or `this.buildStatusUrl` is defined:

```suggestion
        buildStatusConfig: this.buildStatusContext !== undefined || this.buildStatusUrl !== undefined
          ? {
            context: this.buildStatusContext,
            targetUrl: this.buildStatusUrl,
          }
          : undefined,
```

This is why the integration test is failing in the build right now:

```
@aws-cdk/aws-codebuild: Verifying integ.github-webhook-batch.js against integ.github-webhook-batch.expected.json ... CHANGED.
@aws-cdk/aws-codebuild: Resources
@aws-cdk/aws-codebuild: [~] AWS::CodeBuild::Project MyProject39F7B0AE 
@aws-cdk/aws-codebuild:  â””â”€ [~] Source
@aws-cdk/aws-codebuild:      â””â”€ [+] Added: .BuildStatusConfig
@aws-cdk/aws-codebuild: Verifying integ.github.js against integ.github.expected.json ... CHANGED.
@aws-cdk/aws-codebuild: Resources
@aws-cdk/aws-codebuild: [~] AWS::CodeBuild::Project MyProject39F7B0AE 
@aws-cdk/aws-codebuild:  â””â”€ [~] Source
@aws-cdk/aws-codebuild:      â””â”€ [+] Added: .BuildStatusConfig
```